UAF-2104 : Retrieve checks for Cash Receipt from DB to display Check Details tab

jreeseUA opened the pull request:
UAF-2104 : Added method to retrieve current checks for Cash Receipt document from the DB so that Check Details tab would display

sskinner commented on kfs-core/src/main/java/edu/arizona/kfs/fp/document/CashReceiptDocument.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.fp.businessobject.Check;
```
There are 23 unused imports here which should be removed (Eclipse hint: ctrl+shift+o will auto-organize and remove dead imports).

sskinner commented on kfs-core/src/main/java/edu/arizona/kfs/fp/document/CashReceiptDocument.java:
```diff
@@ -0,0 +1,83 @@
+     *
+     * @return a map with a representation of the proper primary key
+     */
+    protected Map getCheckDetailskPrimaryKey() {
```
We should always parameterize our method signatures and variables; in this case we would like to perhaps see Map<String, String> being used.

sskinner commented on kfs-core/src/main/java/edu/arizona/kfs/fp/document/CashReceiptDocument.java:
```diff
@@ -0,0 +1,83 @@
+     * @return the collection of checks for this cash receipt document
+     */
+    protected List<Check> retrieveChecks() { 	
+    	return (List<Check>) SpringContext.getBean(BusinessObjectService.class).findMatchingOrderBy(CheckBase.class, getCheckDetailskPrimaryKey(), "financialDocumentDepositLineNumber", true);
```
Magic string "financialDocumentDepositLineNumber" can be pulled from KFSPropertyConstants.FINANCIAL_DOCUMENT_DEPOSIT_LINE_NUMBER.

sskinner commented on kfs-core/src/main/java/edu/arizona/kfs/fp/document/CashReceiptDocument.java:
```diff
@@ -0,0 +1,83 @@
+    protected Map getCheckDetailskPrimaryKey() {
+        Map pk = new HashMap();
+        pk.put("documentNumber", this.getDocumentNumber());
+        pk.put("financialDocumentTypeCode", CashReceiptDocument.DOCUMENT_TYPE);
```
Magic string "financialDocumentTypeCode' can be pulled from KFSPropertyConstants.FINANCIAL_DOCUMENT_TYPE_CODE.

sskinner commented on kfs-core/src/main/resources/edu/arizona/kfs/fp/ojb-fp.xml:
```diff
@@ -1,243 +1,352 @@
 <descriptor-repository version="1.0">
 
-  <!-- DV needs a new class to accept,  sourceAccountingLines use new Accounting Line class -->
```
Looks like the IDE's formatter got a hold of this file, i.e., some places with mixed 2 and 4 spaced indentations have been replaced by 4 spaces. This makes it difficult to diff depending on the tool being used. Generally we would want to leave formatting alone so we can more easily see the meaningful changes. (An exception to this is initial commit of the file addition, which is when we'd want to apply formatting to the entire file).

sskinner commented on kfs-core/src/main/resources/edu/arizona/kfs/fp/ojb-fp.xml:
```diff
@@ -1,243 +1,352 @@
+	    <field-descriptor name="totalConfirmedCheckAmount" column="FDOC_CNFRMD_CHCK_AMT" conversion="org.kuali.rice.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion" jdbc-type="DECIMAL" />
+	    <field-descriptor name="totalConfirmedCoinAmount" column="FDOC_CNFRMD_COIN_AMT" conversion="org.kuali.rice.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion" jdbc-type="DECIMAL" />
+	    <!--
+	        <field-descriptor name="totalChangeRequestOutAmount" column="FDOC_CHANGEOUT_AMT" conversion="org.kuali.rice.core.framework.persistence.ojb.conversion.OjbKualiDecimalFieldConversion" jdbc-type="DECIMAL" />
```
Commented-out sections should always be removed.

sskinner commented on kfs-core/src/main/resources/edu/arizona/kfs/fp/ojb-fp.xml:
```diff
@@ -1,243 +1,352 @@
+	    <field-descriptor name="objectId" column="OBJ_ID" jdbc-type="VARCHAR" index="true" />
+	    <field-descriptor name="nextSourceLineNumber" column="FDOC_NXT_LINE_NBR" jdbc-type="INTEGER" />
+	
+		<!-- TODO Why do we need persisting these total amounts? We never populate these fields, their getters always compute values instead of reading from DB. -->
```
It's generally bad form to keep TODO's in newly committed code. Moreover in this case the comment no longer even applies, since these values do indeed appear to be set directly on the instance variables, and not computed as the comment claims.

jreeseUA commented:
Removed unused imports, replaced string literals with KFSPropertyConstants where existing, and parameterized method signatures in edu.arizona.kfs.fp.document.CashReceiptDocument.   Restored indentation formatting from respective code sources, removed commented code block, removed TODO comment in edu.arizona.kfs.fp.ojb-fp.

jreeseUA commented:
Added static typing <String, String> to Map return value and variable of method getCheckDetailskPrimaryKey.

sskinner commented:
Looks clean, clear, and concise; a pleasure to look at now, and will be a pleasure to maintain in the future. Great job!
